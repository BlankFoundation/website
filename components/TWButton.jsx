import classMapMerge from '../lib/classMapMerge'

const TWButton = (props) => {
  const { classMap, ...reactProps } = props;

  return (
    <button
      className={
        classMapMerge(
          {
            fontColor: 'text-white',
            background: 'bg-green-500',
            padding: 'px-3 py-1'
          },
          classMap
        )
      }
      { ...reactProps }
    >
      {props.children}
    </button>
  );
}

TWButton.defaultProps = {
  
  children: "Button",
  title: "Blank.Foundation",
  social: [{"href":"https://discord.gg/HZM2jcHdEU","name":"Discord"},{"href":"https://twitter.com/blankfdn","name":"Twitter"},{"href":"https://medium.com/@blankfdn","name":"Medium"},{"href":"https://www.notion.so/blankfoundation/Blank-Home-f3fd5330df0d4933aa41e363719c2625","name":"Notion"}]
}

export default TWButton;