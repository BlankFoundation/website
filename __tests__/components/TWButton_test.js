import React from 'react';
import TestRenderer from 'react-test-renderer';
import TWButton from '../../components/TWButton';

describe("TWButton", () => {
  it('renders correctly', () => {
    const rendered = TestRenderer.create(<TWButton />).toJSON();
    expect(rendered).toMatchSnapshot()
  });
});