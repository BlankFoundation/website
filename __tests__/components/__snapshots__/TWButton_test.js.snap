// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TWButton renders correctly 1`] = `
<button
  className="text-white bg-green-500 px-3 py-1"
>
  Button
</button>
`;
